### ISSUE TITLE
`encode!': can't modify frozen String when writing character to temporary file

### ISSUE THREAD

**Issue by Mogztter** [oracle/truffleruby#1714]
**test.rb**
```rb
# frozen_string_literal: true
require 'tempfile'

tmp_input = Tempfile.new %w(test- .adoc), encoding: Encoding::IBM437
# NOTE using a character whose code differs between UTF-8 and IBM437
tmp_input.write %(ƒ\n)
tmp_input.close
```

```console
$ ruby -v     
truffleruby 19.1.0, like ruby 2.6.2, GraalVM CE Native [x86_64-linux]
```

```console
$ ruby test.rb --trace
/home/guillaume/.rvm/rubies/truffleruby-19.1.0/lib/mri/delegate.rb:349:in `encode!': can't modify frozen String (FrozenError)
	from /home/guillaume/.rvm/rubies/truffleruby-19.1.0/lib/mri/delegate.rb:349:in `write'
	from /home/guillaume/.rvm/rubies/truffleruby-19.1.0/lib/mri/delegate.rb:349:in `write'
	from test.rb:6:in `<main>'

```

**Comment by eregon** [comment]
I fixed this in 7df477321b36374927d7d1a1199e50fa58d556b1, it will be in the next release.
@Mogztter Thank you for reporting the issue!

**Comment by eregon** [comment]
TruffleRuby 19.2.0 was released and fixes this bug, please try it.
